presto-main/src/test/java/com/facebook/presto/operator/aggregation/noisyaggregation/TestNoisySumGaussianDoubleAggregation.java

```diff
@@ -0,0 +1,419 @@
+                noisySumGaussian,
+                notEqualDoubleAssertion,
+                "Test noisy_sum_gaussian(double, noiseScale) with noiseScale > 0 which means some noise",
+                new Page(
+                        createDoublesBlock(values),
+                        createRLEBlock(DEFAULT_TEST_STANDARD_DEVIATION, numRows)),
+                expected);
+    }
+
+    @Test
+    public void testNoisySumGaussianDoubleSomeNoiseScaleWithinSomeStd()
+    {
+        JavaAggregationFunctionImplementation noisySumGaussian = getFunction(DOUBLE, DOUBLE);
+
+        BiFunction<Object, Object, Boolean> withinSomeStdAssertion = (actual, expected) -> {
+            double actualValue = new Double(actual.toString());
+            double expectedValue = new Double(expected.toString());
+            return expectedValue - 50 * DEFAULT_TEST_STANDARD_DEVIATION <= actualValue && actualValue <= expectedValue + 50 * DEFAULT_TEST_STANDARD_DEVIATION;
+        };
+
+        int numRows = 1000;
+        List<Double> values = createTestValues(numRows, false, 1.0, true);
+        double expected = sum(values);
+        assertAggregation(
+                noisySumGaussian,
+                withinSomeStdAssertion,
+                "Test noisy_sum_gaussian(double, noiseScale) within some std from mean",
+                new Page(
+                        createDoublesBlock(values),
+                        createRLEBlock(DEFAULT_TEST_STANDARD_DEVIATION, numRows)),
+                expected);
+    }
+
+    // Test DOUBLE vs. normal SUM
+    @Test
+    public void testNoisySumGaussianDoubleNoiseScaleVsNormalSum()
+    {
+        // Test DOUBLE SUM(col) producing the same values
+
+        int numRows = 10;
+        String data = buildData(numRows, true, Arrays.asList(
+                StandardTypes.BIGINT,
+                StandardTypes.DOUBLE,
+                StandardTypes.REAL,
+                StandardTypes.DECIMAL));
+        String columnName = buildColumnName(StandardTypes.DOUBLE);
+        String query1 = String.format("SELECT SUM(%s) FROM %s", columnName, data);
+        String query2 = String.format("SELECT %s(%s, %f) FROM %s", FUNCTION_NAME, columnName, 0.0, data);
+
+        List<MaterializedRow> actualRows = runQuery(query1);
+        double result1 = Double.parseDouble(actualRows.get(0).getField(0).toString());
+
+        actualRows = runQuery(query2);
+        double result2 = Double.parseDouble(actualRows.get(0).getField(0).toString());
+
+        assertEquals(result2, result1);
+    }
+
+    // Test DOUBLE with clipping
+    @Test
+    public void testNoisySumGaussianDoubleClippingZeroNoiseScale()
+    {
+        JavaAggregationFunctionImplementation noisySumGaussian = getFunction(DOUBLE, DOUBLE, DOUBLE, DOUBLE);
+
+        int numRows = 10;
+        List<Double> values = createTestValues(numRows, false, 1.0, false);
+        double lower = 2.0;
+        double upper = 8.0;
+        double expected = clippedSum(values, lower, upper);
```

Comment:
I think I'd prefer to hardcode this, since `clippedSum` calls out to the very functionality being tested (`NoisySumAggregationUtils.clip`).

I think the answer is 53, but don't quote me on that. 😄

Reply:
Guess what. It is 45. The list is [0, ..., 9] though

Reply:
LOL ❤️

Reply:
See. I was wrong as well. Other lines like this in this file are 45 because they have NULL instead of 0:
- When there is NULL, it is [1, ..., 9, NULL], then 1 is clipped to 2 and 9 is clipped to 8, so we still have 45
- When there is no NULL, it is [0, 1, ..., 9], then 0 -> 2,  1 -> 2, 9 -> 8, so we have 47 
:))